File: elixir/medium/2610/2610.exs
defmodule Solution do
  @spec find_matrix(nums :: [integer]) :: [[integer]]
  def find_matrix(nums) do
    counter = Enum.frequencies(nums) |> IO.inspect()
    matrix = Enum.map(counter, fn {k, v} -> List.duplicate(k, v) end) |> IO.inspect()
    Transp.transpose(matrix) |> IO.inspect()
  end
end

defmodule Transp do
  def transpose([]), do: []
  def transpose([[] | xss]), do: transpose(xss)

  def transpose([[x | xs] | xss]) do
    [[x | for([h | _t] <- xss, do: h)] | transpose([xs | for([_h | t] <- xss, do: t)])]
  end
end

File: py/medium/2610/2610.py
from collections import Counter
from typing import List


class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        counter = Counter(nums)
        matrix = [[] for _ in range(max(counter.values()))]
        for key, value in counter.items():
            for i in range(value):
                matrix[i].append(key)
        return matrix


solution = Solution()
found_matrix = solution.findMatrix(nums=[1, 3, 4, 1, 2, 3, 1])
test_matrix = [[1, 3, 4, 2], [1, 3], [1]]
assert [set(x) for x in found_matrix] == [set(x) for x in test_matrix]

found_matrix = solution.findMatrix(nums=[1, 2, 3, 4])
test_matrix = [[4, 3, 2, 1]]
assert [set(x) for x in found_matrix] == [set(x) for x in test_matrix]
